src/warc2text_runner/two/fastertext_lid/proto_langid.py

```diff
@@ -4,66 +4,116 @@
+        python -m langid_scripts.proto_langid --model_path $MODEL_PATH < $YOUR_FILE
 
         """
+        if use_logging is True:
+            self.logger = langid_logger(name="basic_langid_logger", level=level_log)
+        else:
+            self.logger = logging.getLogger(name="basic_langid_logger_disabled")
+            self.logger.disabled = True
+
         self.model = fasttext.load_model(model_path)
+        self.logger.debug("FastTextLangId model loaded.")
 
     def _preproccess_text(self, text: str) -> str:
-        """Preprocesses the text. Naive cleaning."""
-        return text.replace("\n", " ").strip()
+        """Preprocesses a single line of text for lang ID."""
+        if not isinstance(text, str):
+            msg = "Input text must be a string."
+            raise TypeError(msg)
+
+        self.logger.debug("Before: %s", text)
+        text = regex.sub(NONWORD_REPLACE_PATTERN, "", text.replace("\n", " ").strip())
+        self.logger.debug("After: %s", text)
+        return text
+
+    def _postprocess_predicted_labels(self, prediction: tuple) -> list[str]:
+        """
+        Postprocess the predicted labels.
+
+        Example: "__label__eng_Latn" -> "eng_Latn"
+        """
+        return [label[9:] for label in prediction[0]]
+
+    def _postprocess_predicted_probabilities(self, prediction: tuple) -> list[float]:
+        """
+        Postprocess the predicted probabilities.
 
-    def _postprocess_prediction(self, prediction: tuple) -> str:
-        """Postprocesses the prediction."""
-        return prediction[0][0].split("__label__")[1]
+        Example: [0.92134414] -> [0.9213]
+        """
+        rounded_probs = numpy.round(prediction[1], decimals=4)
+
+        return rounded_probs.tolist()
 
     def predict_language_from_stdin_jsonlines(self) -> None:
         """
         Read from stdin jsonlines.
 
-        Expected input: {"t":"text"}
-        Expected output: {"lang":"en", "prob":0.9}
+        Example input:
+
+        {"t": "Hello, world!"}
 
-        Expected output(if None/null it "t" field): {"lang":null}
-        Expected output(if empty string in "t" field): {"lang":"unk"}
+        Example output:
+
+        {"lang": ["eng_Latn"], "prob": [0.9213]}
 
         """
         with fileinput.input(files=("-",), encoding="utf-8") as f:
             for fileinput_line in f:
-                json_line = json.loads(fileinput_line)
+                self.logger.debug("Read fileinput line: %s", fileinput_line)
+                # load json line
+                json_line = ujson.loads(fileinput_line)
+                self.logger.debug("Read json line.")
 
                 if json_line["t"] is None:
-                    print(json.dumps({'lang': None}))
-                elif json_line["t"] == "":
-                    print(json.dumps({'lang': '_unk'}))
+                    self.logger.debug("Case: text is None.")
+                    print(ujson.dumps({"lang": ["_null"]}))
```

Comment:
I would suggest using Python None here as the output for None input. It will simplify the code handling that: e.g. we will be able to write ```if json_line.get("lang"): ...``` instead of ```if json_line.get("lang")=="_null"```

Reply:
I agree, we should reduce the possible values of null.

Reply:
Is this works?

Now:
```
{"lang":["langcode1", "langcode2", "langcode3"],"prob":[0.9975,0.0004,0.0004]} # normal case
{"lang":["_null"]} # get None from "t" field
{"lang":["_unk"]} # get empty string from "t" field
```

Will be:
```
{"lang":["langcode1", "langcode2", "langcode3"],"prob":[0.9975,0.0004,0.0004]}
{"lang":null}
{"lang":["_unk"]}
```